src/crush/CrushWrapper.cc

```diff
@@ -1647,12 +1647,22 @@ int CrushWrapper::update_device_class(int id,
                                       const string& name,
                                       ostream *ss)
 {
+  assert(item_exists(id));
+  auto old_class_name = get_item_class(id);
+  stringstream ts;
+  if (old_class_name &&
+      old_class_name != class_name && // so command is idempotent
+      class_is_in_use(get_class_id(old_class_name), &ts)) {
+    *ss << "osd." << id << " has already bound to class '" << old_class_name
+        << "', which is currently " << ts.str();
+    return -EBUSY;
```

Comment:
I'm not sure I understand the logic here.  If the class is wrong, what is the user supposed to do?